src/coreclr/src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComActivator.cs

```diff
@@ -267,6 +267,7 @@ public static unsafe int GetClassFactoryForTypeInternal(ComActivationContextInte
             {
                 var cxt = ComActivationContext.Create(ref cxtInt);
                 object cf = GetClassFactoryForType(cxt);
+                Debug.Assert(OperatingSystem.IsWindows());
```

Comment:
Since this is COM - would it be more appropriate to mark this whole class as only supported on Windows?

Reply:
Will annotate come related ones as windows only, thanks